akka-actor/src/main/scala/akka/actor/FaultHandling.scala

```diff
@@ -177,6 +178,18 @@ object SupervisorStrategy extends SupervisorStrategyLowPriorityImplicits {
     }
     OneForOneStrategy()(stoppingDecider)
   }
+  
+  final def detectAkkaBinaryIncompatibilityWarningAndStop(m: IncompatibleClassChangeError): Directive = {
+    if (m.getMessage startsWith "akka") 
+      println(
+        s"""!!! Detected ${m.getClass} error, which MAY be caused by incompatible Akka versions on the classpath.       !!! 
+           |!!! Please note that a given Akka version MUST be the same across all modules of Akka that you are using,   !!!
+           |!!! e.g. if you use akka-actor [${akka.Version.current} (resolved from current classpath)] all other core   !!!
+           |!!! Akka modules MUST be of the same version. External projects like Alpakka, Persistence plugins or Akka   !!!
+           |!!! HTTP etc. have their own version numbers - please make sure you're using a compatible set of libraries. !!!
+         """.stripMargin)
```

Comment:
Log messages should be one line, no newlines. I don't think all these exclamation marks are needed.